app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -1087,6 +901,81 @@ private fun creditCardSuggestion(creditCardNumber: String) =
 private fun siteSecurityToolbarButton() =
     mDevice.findObject(UiSelector().resourceId("$packageName:id/mozac_browser_toolbar_security_indicator"))
 
+private fun clickPageObject(webPageItem: UiObject) {
+    for (i in 1..RETRY_COUNT) {
+        try {
+            webPageItem.also {
+                it.waitForExists(waitingTime)
+                it.click()
+            }
+
+            break
+        } catch (e: UiObjectNotFoundException) {
+            if (i == RETRY_COUNT) {
+                throw e
+            } else {
+                browserScreen {
+                }.openThreeDotMenu {
+                }.refreshPage {
+                }
+                progressBar.waitUntilGone(waitingTime)
+            }
+        }
+    }
+}
+
+fun longClickPageObject(webPageItem: UiObject) {
+    for (i in 1..RETRY_COUNT) {
+        try {
+            webPageItem.also {
+                it.waitForExists(waitingTime)
+                it.longClick()
+            }
+
+            break
+        } catch (e: UiObjectNotFoundException) {
+            if (i == RETRY_COUNT) {
+                throw e
+            } else {
+                browserScreen {
+                }.openThreeDotMenu {
+                }.refreshPage {
+                }
+                progressBar.waitUntilGone(waitingTime)
+            }
+        }
+    }
+}
+
+private fun webPageItemContainingText(itemText: String) =
+    mDevice.findObject(UiSelector().textContains(itemText))
+
+private fun webPageItemWithText(itemText: String) =
+    mDevice.findObject(UiSelector().text(itemText))
+
+private fun setPageObjectText(webPageItem: UiObject, text: String) {
+    for (i in 1..RETRY_COUNT) {
+        try {
+            webPageItem.also {
+                it.waitForExists(waitingTime)
+                it.setText(text)
+            }
+
+            break
+        } catch (e: UiObjectNotFoundException) {
+            if (i == RETRY_COUNT) {
+                throw e
+            } else {
+                browserScreen {
+                }.openThreeDotMenu {
+                }.refreshPage {
+                }
+                progressBar.waitUntilGone(waitingTime)
```

Comment:
nit: progressBar.waitUntilGone(waitingTime) would look better between the refreshPage curly braces

Reply:
Sounds good 😝